### PR TITLE
Git integration documentation

### LINKED ISSUE #446: Docs: Add page for the Git features
### Before you start...

- [X] Have you refreshed the page? You might be using an old version of the website.
- [X] Have you checked if someone already requested this feature? If they have, post a comment with your use case instead.

### Description

We added a number of Git related features in v2.5. Besides helping users with the feature, documentation will help with SEO as well.

### Alternative solutions

_No response_

### Relevant Assets

_No response_

### PR DESCRIPTION
**Pull request by oleklukasiewicz**
## Description

Initial documentation for git integration.

Closes #446

### CODE REVIEW

**Review by chingucoding — APPROVED** [review]
Awesome, thanks for adding this! 

@yaira2 LGTM, feel free to merge this if you also think this is ready.

File: src/routes/docs/configuring/git-integration/+page.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+---
+title: Manage repositories with Git
+---
+
+Integration with Git allows you to work more easily with your projects. You can create and manage repository branches directly from Files. You can also pull changes from a remote repository and open a repository in VS Code.
```
Comment (yaira2) [inline src/routes/docs/configuring/git-integration/+page.md:5]:
```suggestion
Integration with Git allows you to work more easily with your projects. You can create and manage branches directly from Files. You can pull and push changes to remote repos, and open them in VS Code.```

File: src/routes/docs/configuring/git-integration/+page.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
```
Comment (yaira2) [inline src/routes/docs/configuring/git-integration/+page.md:6]:
```suggestion

## Using Git in Files

No additional setup is required, opening a Git folder in Files will automatically detect and display the branch name on the Status Bar.

### Switching branches

Clicking the branch name will open a flyout with the list of available branches, you'll also be able to create a new branch from this flyout.
```

File: src/routes/docs/configuring/git-integration/+page.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+Clicking the branch name will open a flyout with the list of available branches, you'll also be able to create a new branch from this flyout.
+## How to see repository branches
+
+To see repository branches, follow these steps:
+
+- Navigate to the folder containing the repository.
+- Click the name of the current branch in the bottom right corner.
+
+You will see a list of all local and remote repository branches.
```
Comment (yaira2) [inline src/routes/docs/configuring/git-integration/+page.md:21]: I think this is covered by the "switching branches" section.

File: src/routes/docs/configuring/git-integration/+page.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+## How to create a new branch
```
Comment (yaira2) [inline src/routes/docs/configuring/git-integration/+page.md:23]:
```suggestion
### How to create a new branch
```

File: src/routes/docs/configuring/git-integration/+page.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+- Enter the name of the new branch, select the base of the new branch and choose whether you want to switch to the new branch after creation.
+- Click the "Create" button.
+
+## How to switch to another branch
+
+To switch to another branch, follow these steps:
+
+- Navigate to the folder containing the repository.
+- Click the name of the current branch in the bottom right corner.
+- Click the branch you want to switch to in the menu that appears.
```
Comment (yaira2) [inline src/routes/docs/configuring/git-integration/+page.md:39]: This is covered already

File: src/routes/docs/configuring/git-integration/+page.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+## How to pull changes from a remote repository
```
Comment (yaira2) [inline src/routes/docs/configuring/git-integration/+page.md:41]: Can you update this to reflect the changes in the preview builds?

File: src/routes/docs/configuring/git-integration/+page.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+To pull changes from a remote repository, follow these steps:
+
+- Navigate to the folder containing the repository.
+- You will see the number of commits to pull in the bottom right corner next to the branch button.
+- Click the Pull button to pull changes from the remote repository.
+
+## Git columns in the details layout
```
Comment (yaira2) [inline src/routes/docs/configuring/git-integration/+page.md:49]:
```suggestion
### Git columns in the Details layout
```

File: src/routes/docs/configuring/git-integration/+page.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+Opening a Git repository in the details layout displays additional columns for Git status, last commit, commit hash, and commit author. These columns can be toggled by right-clicking the column headers.
+
+## Opening a Git repository in VS Code
+
+When browsing a Git repository, the status bar shows a button that allows you to open a folder in VS Code.
```
Comment (yaira2) [inline src/routes/docs/configuring/git-integration/+page.md:55]:
```suggestion
When browsing a Git repository, the status bar shows a button that allows you to open a folder in VS Code. Opening a folder with a Visual Studio solution file will display an option to open the solution in Visual Studio.
```

File: src/routes/docs/configuring/git-integration/+page.md [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+
+- Navigate to the folder containing the repository.
+- You will see the number of commits to pull in the bottom right corner next to the branch button.
+- Click the Pull button to pull changes from the remote repository.
+
+## Git columns in the details layout
+
+Opening a Git repository in the details layout displays additional columns for Git status, last commit, commit hash, and commit author. These columns can be toggled by right-clicking the column headers.
+
+## Opening a Git repository in VS Code
```
Comment (yaira2) [inline src/routes/docs/configuring/git-integration/+page.md:53]:
```suggestion
### Opening a Git repository in VS Code
```

File: src/routes/docs/configuring/git-integration/+page.md (hunk 6 above)
Comment (oleklukasiewicz) [reply src/routes/docs/configuring/git-integration/+page.md:41]: Sure!

**Review by yaira2 — APPROVED** [review]
LGTM

@oleklukasiewicz thank you for taking care of this!

**Comment by oleklukasiewicz** [comment]
Hey @yaira2, can I take care of the rest of the requested docs for the website?

**Comment by yaira2** [comment]
@oleklukasiewicz that would be great! I'll probably want to edit it like I did with the Git documentation but this would be a big help.

**Pull request merged**